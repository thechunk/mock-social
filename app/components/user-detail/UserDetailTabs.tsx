import * as React from 'react';
import {PureComponent} from 'react';
import {SegmentedControl, TabBar, Tab, Content} from '../segmented-control';
import styles from './styles';
import g, {Color} from '../../styles/global';
import UserDetailPostsTab from "./UserDetailPostsTab";
import UserDetailTodosTab from "./UserDetailTodosTab";

export default class UserDetailTabs extends PureComponent<IUserTabs> {
    render() {
        return (
            <>
                <SegmentedControl>
                    <TabBar style={styles.tabBar}>
                        <Tab
                            inactiveStyle={[styles.tab, {backgroundColor: Color.SecondaryAccent}]}
                            activeStyle={styles.tabActive}
                            inactiveTextStyle={styles.tabTextInactive}
                            activeTextStyle={styles.tabTextActive}
                            id={0}
                            title='Posts' />
                        <Tab
                            inactiveStyle={[styles.tab, styles.tabLeftBorder, {backgroundColor: Color.SecondaryAccent}]}
                            activeStyle={styles.tabActive}
                            inactiveTextStyle={styles.tabTextInactive}
                            activeTextStyle={styles.tabTextActive}
                            id={1}
                            title='To-Dos' />
                    </TabBar>
                    <Content id={0}>
                        <UserDetailPostsTab id={this.props.user.id} />
                    </Content>
                    <Content id={1}>
                        <UserDetailTodosTab id={this.props.user.id} />
                    </Content>
                </SegmentedControl>
            </>
        );
    }
}