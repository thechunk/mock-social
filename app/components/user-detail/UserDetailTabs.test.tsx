import * as React from 'react';
import {shallow} from 'enzyme';
import UserDetailTabs from './UserDetailTabs';

describe('UserDetailTabs', () => {
    it('renders correctly', () => {
        expect(shallow(<UserDetailTabs/>)).toMatchSnapshot();
    });
});