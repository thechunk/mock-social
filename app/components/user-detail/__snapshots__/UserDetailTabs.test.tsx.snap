// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailTabs renders correctly 1`] = `
<Fragment>
  <SegmentedControl>
    <TabBar
      style={
        Object {
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "paddingLeft": 12,
          "paddingRight": 12,
        }
      }
    >
      <Tab
        activeStyle={
          Object {
            "borderColor": "rgba(0, 0, 0, 0.3)",
          }
        }
        activeTextStyle={
          Object {
            "fontWeight": "bold",
          }
        }
        id={0}
        inactiveStyle={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#c06171",
              "borderBottomLeftRadius": 20,
              "borderBottomRightRadius": 20,
              "borderColor": "transparent",
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "borderWidth": 2,
              "flex": 1,
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#c06171",
            },
          ]
        }
        inactiveTextStyle={
          Object {
            "color": "#fff",
          }
        }
        title="Posts"
      />
      <Tab
        activeStyle={
          Object {
            "borderColor": "rgba(0, 0, 0, 0.3)",
          }
        }
        activeTextStyle={
          Object {
            "fontWeight": "bold",
          }
        }
        id={1}
        inactiveStyle={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#c06171",
              "borderBottomLeftRadius": 20,
              "borderBottomRightRadius": 20,
              "borderColor": "transparent",
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "borderWidth": 2,
              "flex": 1,
              "height": 40,
              "justifyContent": "center",
            },
            Object {
              "marginLeft": 12,
            },
            Object {
              "backgroundColor": "#c06171",
            },
          ]
        }
        inactiveTextStyle={
          Object {
            "color": "#fff",
          }
        }
        title="To-Dos"
      />
    </TabBar>
    <Content
      id={0}
    >
      <Component
        id={1}
      />
    </Content>
    <Content
      id={1}
    >
      <Component
        id={1}
      />
    </Content>
  </SegmentedControl>
</Fragment>
`;
